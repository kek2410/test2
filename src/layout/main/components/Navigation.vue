<template>
  <v-navigation-drawer v-model="drawer" app dark>
    <v-subheader>{{ fullName }}(12345)</v-subheader>
    <v-divider />
    <v-card class="mx-auto" max-width="300" tile>
      <v-list v-model="item" dense nav>
        <div v-for="(data, i) in menuList" :key="i">
          <NavigationGroupList v-if="data.group" :data="data" />
          <NavigationList v-else :data="data" />
        </div>
      </v-list>
    </v-card>
  </v-navigation-drawer>
</template>

<script lang="ts">
import { Vue, Prop, Component } from "vue-property-decorator";
import mainRoute from "@/router/main";
import NavigationList from "./NavigationList.vue";
import NavigationGroupList from "./NavigationGroupList.vue";
import { Getter } from "vuex-class";

@Component({
  components: {
    NavigationList,
    NavigationGroupList
  }
})
export default class Navigation extends Vue {
  @Prop() data!: any;
  @Prop() drawer!: any;
  @Getter("fullName") fullName!: string;
  mini = true;
  groupValue = false;
  item = 0;

  menuList: any = null;

  created() {
    this.menuList = mainRoute;
    for (const key in this.menuList) {
      if (this.menuList.hasOwnProperty(key)) {
        const obj = this.menuList[key];
        if (obj.hasOwnProperty("hidden") && obj.hidden) {
          this.menuList.splice(key, 1);
        } else {
          if (obj.hasOwnProperty("children")) {
            this.menuList[key].group = true;
          } else {
            this.menuList[key].group = false;
          }
        }
      }
    }

    this.menuList.sort(function(a: any, b: any) {
      return a.sort < b.sort ? -1 : a.sort > b.sort ? 1 : 0;
    });
  }

  mounted() {}
}
</script>
