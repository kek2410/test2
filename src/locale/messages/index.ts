const getFiles = require.context("./languages", true, /\.ts$/);

const getModules = getFiles
  .keys()
  .reduce((modules: any, modulePath: any): any => {
    let moduleName: any = modulePath;
    moduleName = moduleName.replace(/^\.\/(.*)\.\w+$/, "$1");
    moduleName = moduleName.replace("/index", "");
    const value: any = getFiles(modulePath);
    modules[moduleName] = value.default;
    return modules;
  }, {});

export default getModules;
