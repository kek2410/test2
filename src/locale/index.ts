import Vue from "vue";
import VueI18n from "vue-i18n";
import messages from "./messages";

/* Vuetify 에서 제공해주는 디폴트 언어
  af - Afrikaans (Afrikaans)
  ar - Arabic (اللغة العربية)
  ca - Catalan (català)
  cs - Czech (čeština)
  de - German (Deutsch)
  el - Greek (Ελληνικά)
  en - English
  es - Spanish (Español)
  et - Estonian (eesti)
  fa - Persian (فارسی)
  fr - French (Français)
  he - Hebrew (עברית)
  hr - Croatian (hrvatski jezik)
  hu - Hungarian (magyar)
  id - Indonesian (Indonesian)
  it - Italian (Italiano)
  ja - Japanese (日本語)
  ko - Korean (한국어)
  lt - Lithuanian (lietuvių kalba)
  lv - Latvian (latviešu valoda)
  nl - Dutch (Nederlands)
  no - Norwegian (Norsk)
  pl - Polish (język polski)
  pt - Portuguese (Português)
  ro - Romanian (Română)
  ru - Russian (Русский)
  sl - Slovene (slovenski jezik)
  srCyrl - Serbian (српски језик)
  sv - Swedish (svenska)
  th - Thai (ไทย)
  tr - Turkish (Türkçe)
  uk - Ukrainian (Українська)
  zhHans - Chinese (中文)
  zhHant - Chinese (中文)
*/

Vue.use(VueI18n);

export const seti18n = {
  locale: "ko",
  messages
};

export default new VueI18n(seti18n);
