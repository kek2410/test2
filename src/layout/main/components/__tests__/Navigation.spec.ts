import { shallowMount, createLocalVue } from "@vue/test-utils";
import Navigation from "../Navigation.vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
import VueI18n from "vue-i18n";
import { seti18n } from "@/locale";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueI18n);
localVue.use(Vuetify);

describe("Navigation.vue", (): void => {
  let store: any;
  let vuetify: any;
  let i18n: any;

  beforeEach((): void => {
    i18n = new VueI18n(seti18n);
    vuetify = new Vuetify();
    store = new Vuex.Store({
      getters: {
        menuList: (state: any): any => state.login.menuList
      },
      modules: {
        login: {
          namespaced: true,
          state: {
            menuList: [
              {
                name: "DashBoard",
                icon: "mdi-view-dashboard",
                title: "DashBoard"
              },
              {
                name: "TestList",
                icon: "mdi-home",
                title: "TestList"
              }
            ]
          }
        }
      },
      strict: true
    });
  });

  it("Navigation.vue > test.name1", (): void => {
    const wr = shallowMount(Navigation, {
      localVue,
      store,
      vuetify,
      i18n
    });
    expect(wr.html()).toMatchSnapshot();
  });

  it("Navigation.vue > test.name", async (): Promise<any> => {
    const wr = shallowMount(Navigation, {
      localVue,
      store,
      vuetify,
      i18n
    });

    console.log(wr);

    expect(1).toBe(1);
  });
});
