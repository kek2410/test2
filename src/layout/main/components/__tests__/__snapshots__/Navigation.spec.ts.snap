// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.vue Navigation.vue > test.name1 1`] = `
<div>
  <v-navigation-drawer-stub app="true" height="100vh" minivariantwidth="80" mobilebreakpoint="1264" src="" tag="nav" width="256"></v-navigation-drawer-stub>
</div>
`;
